### PR TITLE
Redesigned person & group pages

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Implements new person page layout based on a [design](https://github.com/PostHog/posthog/issues/7776#issuecomment-996906613) by @clarkus. Closes #7928.

Also added a person 404 page instead of the confusing eternal loading screen that shows up now.

Screen Shot 2022-01-11 at 19 55 36

### CODE REVIEW

**Review by macobo — CHANGES_REQUESTED** [review]
We should keep the groups page design in sync with person page. In this case we're diverging these two pages due to an oversight - please keep this feature in mind going forward!

**Comment by clarkus** [comment]
Somewhat related https://github.com/PostHog/posthog/issues/8011

**Comment by Twixes** [comment]
Good point @macobo. Reworked the group page too. See person + group flow:

File: frontend/src/scenes/groups/groupLogic.ts [hunk 1]
```diff
@@ -40,28 +40,21 @@ export const groupLogic = kea<groupLogicType>({
...
-            (groupTypeName, groupTypeIndex, groupKey): Breadcrumb[] => [
+            (s) => [s.groupTypeName, s.groupTypeIndex, s.groupKey, s.groupData],
+            (groupTypeName, groupTypeIndex, groupKey, groupData): Breadcrumb[] => [
                 {
                     name: capitalizeFirstLetter(groupTypeName),
                     path: urls.groups(String(groupTypeIndex)),
                 },
                 {
-                    name: groupKey,
+                    name: groupDisplayId(groupKey, groupData?.group_properties || {}),
```
Comment (macobo) [inline frontend/src/scenes/groups/groupLogic.ts:57]: Great catch :)

**Review by macobo — APPROVED** [review]
Took a look code-wise! Good job, the changes are easy to follow :)

**Review by paolodamico — APPROVED** [review]
Seeing @macobo reviewed the code, I did a UX review. Some minor improvements but nothing blocking.

1. The alignment of the button to remove props seems strange, I would consider putting at the right most place of the entire table.
Screen Shot 2022-01-18 at 8 36 38 AM
2. Update copy for empty state on recordings (e.g. "This user doesn't have any recordings yet. Old recordings are removed after X weeks from your account.")
3. Cohorts list doesn't get recalculated unless I reload the page. I added a cohort that matched a user, went back to the user page and the cohort didn't show up.

**Comment by clarkus** [comment]
Agree that actions should be in the right most column. We use trashcan icons for removal / delete actions, not `x`. Typically I have been reserving `x` icons for clearing state / removing temporary things in a non-destructive way.

**Comment by Twixes** [comment]
Good points! This is properties now:
Person properties
Only used a slightly different empty recordings message as I _think_ they're not deletable on ClickHouse currently @paolodamico.

**Pull request merged**